Sideview and flowline layers

bastiaan-roos opened the pull request:
Layer creation from netCDF after selection of latest netCDF (only for the first netCDF)
support of this layer in the Graph tool
Sideview prototype for version 2 models, including waterlevels (work in progress)

Commit:
initial version of sideview tool

- button
- UI
- first setup for path between 2 points

Commit:
first working version of sideview (W.I.P)

- route logic moved to seperate file
- fine tuned route logic
- draw sideview profile (limitted up to now)

Commit:
additional instruction for installation on Windows

Commit:
removed wrong part of demonstration code

Commit:
improved naming of parameter fname into filename

Commit:
extent sideview with extra information of manholes and surface and drain level

Commit:
Revert "additional instruction for installation on Windows"

This reverts commit 14e2bf5b0cdf2d79921930c74763bec7507c3c0f.

Commit:
add extra info to readme about windows installation

Commit:
fix bug and check on id_mapping file

Commit:
fixes

- improve check on existing id_mapping
- bugfix on select active model database in combobox of result selection
widget

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into sideview

Commit:
Improved loading of layers after model selection

Commit:
moved id_mapping check to loading of netCDF file

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into sideview

# Conflicts:
#	README.rst
#	views/result_selection.py

Commit:
add timeslider

Commit:
display of water level in sideview

Commit:
add to gitignore

Commit:
add progress bar communication

Commit:
add some stylings

Commit:
Moved layer creation to utils and add extra fields

- moved processing to utils, so it can be used by other tools
- for lines at type, inp id and spatialite ids
- add progress information

Commit:
activate buttons after datasource is selected

Commit:
add flowline memory layer support to graph tool

jackieleng commented on .gitignore:
```diff
@@ -10,3 +10,4 @@ remote_debugger_settings.py
 
 # Test data
 test/data
+remote_debugger_settings.py
```
Is denk ik niet nodig. Dit bestand is toch al gecommit, en het misschien wel handig voor andere PyCharm gebruikers later.

jackieleng commented:
Ik zie meerdere keren 'splt_id' staan; het uitschrijven naar 'spatialite_id' is duidelijker. Misschien is het nog beter om er 'object_id' of iets dergelijks van te maken zodat het duidelijk is dat het verwijst naar het oorspronkelijke feature id van de spatialite tabel waar naar het gemapt is. Maar kijk maar wat duidelijker is.

Commit:
add support for messageBar messages

Commit:
create memory layer after selection of netCDF

Commit:
changed field and parameters to more readable 'spatialite_id'

Commit:
bugfix - spatialite was not always selected correctly

Commit:
switches of auto adding model layers

was to slow because of invalid layers. - find out later why..

Commit:
bugfix/ improvement for detection of netCDF compatibility

need latest version of netCDF. now detect an recent change to check on
compatibility

Commit:
make route compatible with memory layers

- add support of configurable id_fields

Commit:
support of sideview v2

- support of all elements except channels
- creation of a combined layer with all objects and data to draw
sideview
- selection of lines instead of points
- still work in progress

jackieleng committed:
Update gitignore
